### PR TITLE
Resolve #27 by fixing mistakes in the documentation

### LINKED ISSUE #27: Inconsistencies in documentation.
site: https://chrthomsen.github.io/pygrametl/doc/examples/dimensions.html#typeoneslowlychanging-dimension
says: "A slowly changing dimension that is only type 1 is currently not supported." however later it
says: "TypeOneSlowlyChangingDimension allows the creation of a Type 1 slowly changing dimension."

site: https://chrthomsen.github.io/pygrametl/doc/examples/dimensions.html#cacheddimension
The example code shows: `from pygrametl.datasources import CSVSource` however it is unused
The example code shows: `for row in sales:` however sales is not defined anywhere.

site: https://chrthomsen.github.io/pygrametl/doc/quickstart/beginner.html
The example code shows: `region_source = CSVSource(csvfile=region_file_handle, delimiter=',')` however `csvfile` appears renamed to `f` (not a great name) ???

In generel it appears random if imports are included or not in the example code. 

I get that these are _minor_ issues, but when I try to learn a new framework/library it is essential that the documentation, especially the quickstart-guide, doesn't contain: 
- outdated information
- undefined variables
- unused imports

I really like this project and think better documentation could have helped me learn pygrametl faster. 
If this is not the correct forum for this feedback please let me know and I will delete my issue.

Best Regards Jonas

### PR DESCRIPTION
**Pull request by skejserjensen**

### CODE REVIEW

File: docs/examples/dimensions.rst [hunk 1]
```diff
@@ -235,227 +239,213 @@ necessary as the exact way to perform bulk loading differs from DBMS to DBMS.
...
+with a few differences. The primary difference between the two classes besides
+the additional method :meth:`.TypeOneSlowlyChangingDimension.scdensure`, is that
+:class:`.TypeOneSlowlyChangingDimension` always caches rows when they are
+inserted. This is done to minimize the number of round trips to the database
+needed for :meth:`.TypeOneSlowlyChangingDimension.scdensure` to increase its
+throughput. The user must also specify the sequence of attributes to use for
+looking up keys as :attr:`.lookupatts`, and can optionally specify the sequence
+of type 1 slowly changing attributes :attr:`.type1atts`. If :attr:`.type1atts`
+is not given it will default to all attributes minus :attr:`.lookupatts`. The
+sequences :attr:`.lookupatts` and :attr:`.lookupatts` must be disjoint and an
```
Comment (chrthomsen) [inline docs/examples/dimensions.rst:326]: type1atts, right?

File: docs/examples/dimensions.rst [hunk 2]
```diff
@@ -235,227 +239,213 @@ necessary as the exact way to perform bulk loading differs from DBMS to DBMS.
...
-accordingly when a change happens. As opposed to a
-:class:`.SlowlyChangingDimension`, a Type 1 slowly changing dimension does not
-include any history or time stamps, so it is important that the rows are
-introduced in chronological order.
+The above example shows a scenario where the price of a product changes over
+time. The instance of :class:`.TypeOneSlowlyChangingDimension` automatically
+checks if a product already exists, and if it does, updates the price if the old
+and new values differ. As a type 1 slowly changing dimension does store the
+history of the changes only the value of the last row to be inserted will be
+stored, so the rows most bu loaded in chronological order. If the history of the
```
Comment (chrthomsen) [inline docs/examples/dimensions.rst:380]: be

File: docs/examples/dimensions.rst [hunk 3]
```diff
@@ -235,227 +239,213 @@ necessary as the exact way to perform bulk loading differs from DBMS to DBMS.
...
-    # about which attribute holds a time stamp for when the row's validity
-    # starts and ends. The parameter fromfinder is also given, which is must be
-    # set to the function that should be used to compute the time stamp for
-    # when the row becomes valid and given as input the name of the row which
-    # value it should use. In this example, the function datareader from
-    # pygrametl is used which converts time stamp from a string to a Python
-    # datetime.date object to simplify the conversion to the Postgres Date type.
-    productDimension = SlowlyChangingDimension (
+    # This slowly changing dimension is created as type 2 only. Meaning that a
+    # new row only changes the validto attribute in the previous rows. validto
```
Comment (chrthomsen) [inline docs/examples/dimensions.rst:440]: rows -> row ?

File: docs/examples/dimensions.rst (hunk 1 above)
Comment (skejserjensen) [reply docs/examples/dimensions.rst:326]: I agree

File: docs/examples/dimensions.rst (hunk 2 above)
Comment (skejserjensen) [reply docs/examples/dimensions.rst:380]: I agree

File: docs/examples/dimensions.rst (hunk 3 above)
Comment (skejserjensen) [reply docs/examples/dimensions.rst]: I agree

**Pull request merged**